File: api/src/main/java/net/blueshell/api/validation/address/ValidCountryCode.java
package net.blueshell.api.validation.address;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.ElementType.PARAMETER;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({FIELD, PARAMETER})
@Retention(RUNTIME)
@Documented
@Constraint(validatedBy = CountryCodeValidatorImpl.class)
public @interface ValidCountryCode {
    String message() default "Country must be a valid ISO 3166-1 alpha-2 code";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: api/src/main/kotlin/net/blueshell/api/validation/address/ValidCountryCode.kt
package net.blueshell.api.validation.address

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Constraint(validatedBy = [_root_ide_package_.net.blueshell.api.validation.address.CountryCodeValidator::class])
annotation class ValidCountryCode(
    val message: String = "Country must be a valid ISO 3166-1 alpha-2 code",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)
